java/client/src/main/java/glide/api/commands/ListBaseCommands.java

```diff
@@ -56,6 +58,91 @@ public interface ListBaseCommands {
      */
     CompletableFuture<String> lpop(String key);
 
+    /**
+     * Returns the index of matching elements inside a list. By default, it will scan the list from
+     * head to tail, looking for the first match of <code>element</code>. If the <code>element</code>
+     * is found, its index is returned. If no match is found, <code>null</code> is returned.
+     *
+     * @see <a href="https://redis.io/docs/latest/commands/lpos/">redis.io</a> for details.
+     * @param key The name of the list.
+     * @param element The value to search for within the list.
+     * @return The index of <code>element</code>.
+     * @example
+     *     <pre>{@code
+     * assertEquals(6L, client.rpush("my_list", new String[] {"a", "b", "c", "d", "e", "e"}).get());
+     * assertEquals(4L, client.lpos("my_list", "e").get());
+     * }</pre>
+     */
+    CompletableFuture<Long> lpos(String key, String element);
+
+    /**
+     * Returns the index of matching elements inside a list. By default, it will scan the list from
+     * head to tail, looking for the first match of <code>element</code>. If the <code>element</code>
+     * is found, its index is returned. If no match is found, <code>null</code> is returned.
+     *
+     * @see <a href="https://redis.io/docs/latest/commands/lpos/">redis.io</a> for details.
+     * @param key The name of the list.
+     * @param element The value to search for within the list.
+     * @param options The LPos options.
+     * @return The index of <code>element</code>.
+     * @example
+     *     <pre>{@code
+     * assertEquals(6L, client.rpush("my_list", new String[] {"a", "b", "c", "d", "e", "e"}).get());
+     * // Returns the second occurrence of the element "e".
+     * LPosOptions options = LPosOptions.builder().rank(2L).build();
+     * assertEquals(5L, client.lpos("my_list", "e", options);
+     *
+     * // rank and maxLength
+     * LPosOptions options = LPosOptions.builder().rank(1L).maxLength(1000L).build();
+     * assertEquals(4L, client.lpos("my_list", "e", options);
```

Comment:
I'm not sure if this matters or not, technically if we had this complete example in the same place it wouldn't compile because options is declared twice, but since it is an example maybe we don't care and its implied they are separate. Just in case, should we change the variable names to differ?
```suggestion
     * LPosOptions options1 = LPosOptions.builder().rank(2L).build();
     * assertEquals(5L, client.lpos("my_list", "e", options1);
     *
     * // rank and maxLength
     * LPosOptions options2 = LPosOptions.builder().rank(1L).maxLength(1000L).build();
     * assertEquals(4L, client.lpos("my_list", "e", options2);
```